packages/js/product-editor/src/components/product-select/product-select.tsx

```diff
@@ -1,93 +1,173 @@
 /**
  * External dependencies
  */
-import { createElement, Fragment } from '@wordpress/element';
+import { createElement, useEffect, useRef, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import {
-	__experimentalSelectControl as SelectControl,
-	__experimentalSelectControlMenu as Menu,
-	__experimentalSelectControlMenuItem as MenuItem,
-	useAsyncFilter,
-	Spinner,
-} from '@woocommerce/components';
+import { BaseControl, Spinner } from '@wordpress/components';
 import { Product } from '@woocommerce/data';
-import classNames from 'classnames';
+import classnames from 'classnames';
 
 /**
  * Internal dependencies
  */
 import { FormattedPrice } from '../formatted-price';
 import { ProductImage } from '../product-image';
 import { ProductSelectProps } from './types';
+import { ComboboxControl, ComboboxControlOption } from '../combobox-control';
+
+/**
+ * Map the product item to the Combobox core option.
+ *
+ * @param {Product} attr - Product item.
+ * @return {ComboboxControlOption}               Combobox option.
```

Comment:
Is there a reason for the extra spaces?

Reply:
https://github.com/woocommerce/woocommerce/pull/50128/commits/15b233568a699f7bba91f98801af09c93207ef4f